src/lib/ubisys.ts

```diff
@@ -222,4 +222,37 @@ export const ubisysModernExtend = {
 
         return {exposes: [expose], fromZigbee, toZigbee, configure, isModernExtend: true};
     },
+    cover: (): ModernExtend => {
+        // XXX: in the future we should split fromZigbee/toZigbee, onEvent,... and have these as seperate modernExtends
+
+        const exposes: Expose[] = [];
+        exposes.push((device: Zh.Device, options) => {
```

Comment:
@koenkk as talked about on discord, I don't think this will work. Also I'm not sure what type `options` should be here. 

But aside from that, Expose type is not allowed to be a function.

Reply:
Should work with https://github.com/Koenkk/zigbee-herdsman-converters/pull/7463, please check.

Reply:
I rewrote it to this:
```
    cover: (): ModernExtend => {
        // XXX: in the future we should split fromZigbee/toZigbee, onEvent,... and have these as seperate modernExtends

        const dynamicCover: DefinitionExposes = (device: Zh.Device, options: KeyValue): Expose[] => {
            const result: Expose[] = [];
            const coverExpose = e.cover();
            const coverType = (device?.getEndpoint(1).getClusterAttributeValue('closuresWindowCovering', 'windowCoveringType') ?? undefined);
            switch (coverType) { // cf. Ubisys J1 Technical Reference Manual, chapter 7.2.5.1 Calibration
            case 0: // Roller Shade, Lift only
            case 1: // Roller Shade two motors, Lift only
            case 2: // Roller Shade exterior, Lift only
            case 3: // Roller Shade two motors exterior, Lift only
            case 4: // Drapery, Lift only
            case 5: // Awning, Lift only
            case 9: // Projector Screen, Lift only
                coverExpose.withPosition();
                break;
            case 6: // Shutter, Tilt only
            case 7: // Tilt Blind, Tilt only
                coverExpose.withTilt();
                break;
            case 8: // Tilt Blind, Lift & Tilt
            default:
                coverExpose.withPosition().withTilt();
                break;
            }

            result.push(coverExpose);
            result.push(e.power().withAccess(ea.STATE_GET));
            result.push(e.energy().withAccess(ea.STATE_GET));
            result.push(e.linkquality());
            return result;
        };

        return {exposes: dynamicCover, isModernExtend: true};
    },
```

But I can't get it to work.

Reply:
Dropping the last commit, now does with with https://github.com/Koenkk/zigbee-herdsman-converters/pull/7463, which would be my preferred way forward for now.

A modernExtend conversion for src/devices/ubisys.js is on my list of things to do, but preferably not as part of this PR as there is a lot of code here that needs to be refactored.

Reply:
I'm testing this PR, with the last commit dropped now... I'll now how well it works in a bit.